ports/esp32/esp32_rmt.c

```diff
@@ -160,6 +162,36 @@ STATIC mp_obj_t esp32_rmt_max_pulse_length(mp_obj_t self_in) {
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp32_rmt_max_pulse_length_obj, esp32_rmt_max_pulse_length);
 
 
+// Query whether the channel has finished sending pulses. Takes an optional
+// timeout (in ticks of the 80MHz clock), returning true if the pulse stream has
+// completed or false if they are still transmitting (or timeout is reached).
+STATIC mp_obj_t esp32_rmt_wait_done(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_self,    MP_ARG_REQUIRED | MP_ARG_OBJ, {.u_obj = mp_const_none} },
+        { MP_QSTR_timeout, MP_ARG_KW_ONLY  | MP_ARG_INT, {.u_int = 0} },
+    };
+
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    esp32_rmt_obj_t *self = MP_OBJ_TO_PTR(args[0].u_obj);
+
+    esp_err_t err = rmt_wait_tx_done(self->channel_id, args[1].u_int);
```

Comment:
The errors listed on the IDF are 
```
ESP_OK RMT Tx done successfully
ESP_ERR_TIMEOUT Exceeded the ‘wait_time’ given
ESP_ERR_INVALID_ARG Parameter error
ESP_FAIL Driver not installed
```

It would be good to get these errors thrown instead of just `False`

Reply:
I'm not so sure. The API we provide to a MicroPython users shouldn't necessarily be a pass-through.

FAIL can only occur after deinit. I concede that what _should_ happen in this case is an exception is thrown. If I get a chance I'll do that. 

INVALID_ARG shouldn't be able to occur (the channel ought to always be valid, the timeout can't be invalid). 

That only leaves OK and Timeout which map to True/False.

Reply:
Discussed on Slack. I'm happy with this :)